test/functional/test_framework/wallet.py

```diff
@@ -55,25 +55,37 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
+    def sign_tx(self, *, tx):
+        """Sign the tx with witness stack of OP_TRUE"""
+        tx.wit.vtxinwit = [CTxInWitness()]
+        tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        return tx
+
+    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), seq_num = 0xffffffff, from_node, utxo_to_spend=None, existing_tx = None, send_tx = True, default_fee = False):
         """Create and send a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
         self._utxos = sorted(self._utxos, key=lambda k: k['value'])
         utxo_to_spend = utxo_to_spend or self._utxos.pop()  # Pick the largest utxo (if none provided) and hope it covers the fee
+
         vsize = Decimal(96)
-        send_value = satoshi_round(utxo_to_spend['value'] - fee_rate * (vsize / 1000))
+        send_value = satoshi_round(utxo_to_spend['value'] - fee_rate * (vsize / 1000)) if send_tx else satoshi_round(utxo_to_spend['value'] - fee_rate)
```

Comment:
I find this one-line if/else to be pretty confusing. could you just break it into a few lines to clarify the logic flow?

Reply:
I broke it down, I hope it makes more sense now :)